fastn-core/src/library2022/processor/pg.rs

```diff
@@ -266,9 +269,19 @@ async fn execute_query(
     headers: ftd::ast::HeaderValues,
 ) -> ftd::interpreter::Result<Vec<Vec<serde_json::Value>>> {
     let (query, query_args) = super::sql::extract_arguments(query)?;
-
+    let stmt_key = query.to_string();
     let client = pool().await.as_ref().unwrap().get().await.unwrap();
-    let stmt = client.prepare_cached(query.as_str()).await.unwrap();
+
+    // Use a Mutex to ensure only one thread prepares the same statement
+    let mut prepared_statements = PREPARED_STATEMENT.lock().await;
+    let stmt = if !prepared_statements.contains(&stmt_key) {
+        let stmt = client.prepare_cached(query.as_str()).await.unwrap();
+        prepared_statements.insert(stmt_key);
```

Comment:
Clone here, instead of line 272. This was most of the time we will not have to clone.